tools/build/docker/s6/s6-rc.d/lanraragi/finish

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+kill -15 `cat /home/koyomi/lanraragi/public/temp/shinobu.pid-s6`
```

Comment:
Might be interesting to add those kills to the legacy `services.d/lanraragi` in that case?  
I'm curious to know why those are needed too; seems like the old `s6-svscanctl` was doing a fine enough job by just killing the lrr and redis services. 🤔 (Or maybe it wasn't and I never noticed it!)

Reply:
From the docs:
> The supervision tree now resides in /run/service, and you should not attempt to stop it when you want to exit the container; more generally you should not attempt to send control commands to the supervision tree. In particular, you should not try to run the s6-svscanctl -t /var/run/s6/services command - it will not work anyway because the supervision tree has changed locations. If you need to exit a container from the inside, without your CMD dying (or without having declared a CMD), run the /run/s6/basedir/bin/halt command instead. 

Those processes would throw exceptions when redis shutdowns since s6 v3 only kills the main process not the childs so we need to kill those too before redis.